src/core_plugins/i18n/index.js

```diff
@@ -0,0 +1,23 @@
+import * as i18n from './server/i18n';
+
+export default function (kibana) {
+  return new kibana.Plugin({
+    id: 'i18n',
+    config: function (Joi) {
+      return Joi.object({
+        enabled: Joi.boolean().default(true),
+        locale: Joi.string().default('en')
+      }).default();
+    },
+
+    init(server, options) {
+      const defaultLocale = server.config.locale;
```

Comment:
You still have to access the locale via the getter

`const defaultLocale = server.config.get('i18n.locale');`